Moves <immintrin.h> include to after the compile time 'mf16c' check.

diff --git a/renderdoc/3rdparty/half/half.hpp b/renderdoc/3rdparty/half/half.hpp
index ea631eb07..facb1cd77 100644
--- a/renderdoc/3rdparty/half/half.hpp
+++ b/renderdoc/3rdparty/half/half.hpp
@@ -266,9 +266,6 @@
 #if HALF_ENABLE_CPP11_HASH
 	#include <functional>
 #endif
-#if HALF_ENABLE_F16C_INTRINSICS
-	#include <immintrin.h>
-#endif
 
 
 #ifndef HALF_ENABLE_F16C_INTRINSICS
@@ -281,6 +278,10 @@
 	#define HALF_ENABLE_F16C_INTRINSICS __F16C__
 #endif
 
+#if HALF_ENABLE_F16C_INTRINSICS
+	#include <immintrin.h>
+#endif
+
 #ifdef HALF_DOXYGEN_ONLY
 /// Type for internal floating-point computations.
 /// This can be predefined to a built-in floating-point type (`float`, `double` or `long double`) to override the internal 
